logicalplan/plan_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
+
+package logicalplan
+
+import (
+	"testing"
+
+	"github.com/efficientgo/core/testutil"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+func TestDefaultOptimizers(t *testing.T) {
+	cases := []struct {
+		name     string
+		expr     string
+		expected string
+	}{
+		{
+			name:     "common selectors",
+			expr:     `sum(metric{a="b", c="d"}) / sum(metric{a="b"})`,
+			expected: `sum(filter([a="b" c="d"], metric{a="b"})) / sum(metric{a="b"})`,
+		},
+		{
+			name:     "different selectors",
+			expr:     `sum(metric{a="b"}) / sum(metric{c="d"})`,
+			expected: `sum(metric{a="b"}) / sum(metric{c="d"})`,
+		},
+		{
+			name:     "different operator",
+			expr:     `sum(metric{a="b"}) / sum(metric{a=~"b"})`,
+			expected: `sum(metric{a="b"}) / sum(metric{a=~"b"})`,
+		},
```

Comment:
Can we add some test cases for different metric names within the same query?

Reply:
Good point, added 👍